collector/processor/batchprocessor/splitlogs.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package batchprocessor // import "github.com/open-telemetry/otel-arrow/collector/processor/batchprocessor
+
+import (
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+// splitLogs removes logrecords from the input data and returns a new data of the specified size.
```

Comment:
Just curious to understand what happens to the log records that are located after the specified size is exceeded?

Reply:
The current behavior of this code, i.e., what's being forked in this PR by copying the core batchprocessor, is behavior we want to change. Currently, the batch processor can send at most one batch in parallel, so it requires certain downstream behavior (i.e., the exporterhelper's queue sender) to function well.

So, you'll see more about this in a subsequent PR, where we fix the behavior. The high-level picture that's missing from this dump of copied code, is that it uses pdata objects to accumulate points in a FIFO manner. Batches are assembled in a single pending `pdata` object and then when there are enough points/spans/records, when the pending size is too big it will be split by taking the first points out into a new batch, leaving a residual pdata object. We will ensure that the points in the residual are always the first to go next, which means up to a timeout but no more than one timeout for every point.